sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -79,10 +109,14 @@ struct joint_matrix {
 }
 ```
 
-IMPORTANT: Matrix layout defaulting to `layout::dynamic` applies only to matrix with `use::accumulator`
+IMPORTANT: Matrix layout defaulting to `layout::dynamic` applies only
+to `joint_matrix` with `use::accumulator`
 
-#### Use
-Specifying the usage of the matrix: matrix left (A), matrix right (B) or accumulator +(C)+ is required by backend implementations to reason about the layout of the matrix in registers.
+==== Use
+The main operation performed by the matrix hardware is `D=C+A*B`. `Use`
```

Comment:
We don't actually allow D!=C in the interface. So this could be a little confusing unless we clarify this distinction between what hardware can do and what the interface exposes? Should we really say "main operation performed by the matrix extension" and "C=C+A*B" and so on?

Reply:
This seems like an important issue to resolve.  I remember asking early on whether we could really support the case when `D != C`, and I thought the answer was "yes".  I think this was before we decided to support CUDA with this API, though, so perhaps the answer is different for CUDA.

If we cannot support this case, then I think we need to change the API to make it impossible for the application to ask for something that is not supported.  For example, maybe `joint_matrix_mad` should be defined to modify the `C` parameter in-place.  Does `joint_matrix_mad` need to return anything in this case?

Reply:
Sorry I am plain wrong here. I was thinking about the fact that the types of the returned matrix D and the C must be the same joint_matrix type according to the interface. In practice this often means that the usage is that C==D but the API already allows that C!=D. I think it is fine as it is and you can ignore my thoughtless comment.

Reply:
I confirm that we support C!=D for all backends including CUDA.  So this should not be an issue.